libraries/chain/include/eosio/chain/db_util.hpp

```diff
@@ -0,0 +1,97 @@
+#pragma once
+#include <b1/chain_kv/chain_kv.hpp>
+#include <chainbase/chainbase.hpp>
+#include <eosio/chain/authorization_manager.hpp>
+#include <eosio/chain/block_state.hpp>
+#include <eosio/chain/chain_snapshot.hpp>
+#include <eosio/chain/genesis_state.hpp>
+#include <eosio/chain/resource_limits.hpp>
+#include <eosio/chain/snapshot.hpp>
+
+#include <eosio/chain/account_object.hpp>
+#include <eosio/chain/backing_store.hpp>
+#include <eosio/chain/block_summary_object.hpp>
+#include <eosio/chain/code_object.hpp>
+#include <eosio/chain/contract_table_objects.hpp>
+#include <eosio/chain/database_header_object.hpp>
+#include <eosio/chain/generated_transaction_object.hpp>
+#include <eosio/chain/genesis_intrinsics.hpp>
+#include <eosio/chain/global_property_object.hpp>
+#include <eosio/chain/kv_chainbase_objects.hpp>
+#include <eosio/chain/protocol_state_object.hpp>
+#include <eosio/chain/transaction_object.hpp>
+#include <eosio/chain/whitelisted_intrinsics.hpp>
+#include <eosio/chain/controller.hpp>
+
+#include <b1/session/session.hpp>
+
+namespace eosio { namespace chain { 
+   
+   class apply_context;
+
+   namespace backing_store {
+      struct db_context;
+   }
+
+namespace db_util {
+   using controller_index_set =
+         index_set<account_index, account_metadata_index, account_ram_correction_index, global_property_multi_index,
+                   protocol_state_multi_index, dynamic_global_property_multi_index, block_summary_multi_index,
+                   transaction_multi_index, generated_transaction_multi_index, table_id_multi_index, code_index,
+                   database_header_multi_index, kv_db_config_index>;
+
+   using contract_database_index_set = index_set<key_value_index, index64_index, index128_index, index256_index,
+                                                 index_double_index, index_long_double_index>;
+
+   using db_context = backing_store::db_context;
+
+   class maybe_session {
+    public:
+      maybe_session() = default;
+
+      maybe_session(chainbase::database& cb_database);
+      maybe_session(maybe_session&& src) noexcept;
```

Comment:
Make these constructors `explicit`

Reply:
did it for `maybe_session(chainbase::database& cb_database)`
move constructor is not suggested to be declared explicit. any reason for that?
https://stackoverflow.com/questions/6758717/explicit-move-constructor

Reply:
Ah, yes, move constructor, right, only explicit for the non-move constructor.